From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Akihiko Odaki <odaki@rsg.ci.i.u-tokyo.ac.jp>
Date: Mon, 22 Dec 2025 15:48:33 +0900
Subject: [PATCH 2/2] Enable eglGetPlatformDisplay() for 1.5

eglGetPlatformDisplay() may be used to get a display for a rendering
context. However, epoxy_conservative_egl_version() cannot report a
correct version without a display and "conservatively" reports 1.4
instead. This disables eglGetPlatformDisplay(), which is available only
since 1.5.

To fix this issue, query the client version for eglGetPlatformDisplay().
The client version is a concept introduced with 1.5, and does not
require a display to query. Failing the query implies the version is
older than 1.5 and eglGetPlatformDisplay() is not supported.
---
 src/gen_dispatch.py | 17 ++++++++++++-----
 1 file changed, 12 insertions(+), 5 deletions(-)

diff --git a/src/gen_dispatch.py b/src/gen_dispatch.py
index 117c954..dad7b4d 100755
--- a/src/gen_dispatch.py
+++ b/src/gen_dispatch.py
@@ -363,11 +363,7 @@ class Generator(object):
                     condition = 'true'
                     loader = 'epoxy_glx_dlsym({0})'
             elif api == 'egl':
-                human_name = 'EGL {0}'.format(version)
-                if version > 10:
-                    condition = 'epoxy_conservative_egl_version() >= {0}'.format(version)
-                else:
-                    condition = 'true'
+                condition = 'true'
                 # All EGL core entrypoints must be dlsym()ed out --
                 # eglGetProcAdddress() will return NULL.
                 loader = 'epoxy_egl_dlsym({0})'
@@ -381,6 +377,17 @@ class Generator(object):
                 sys.exit('unknown API: "{0}"'.format(api))
 
             for command in feature.findall('require/command'):
+                if api == 'egl':
+                    human_name = 'EGL {0}'.format(version)
+                    if version > 10:
+                        if command.get('name') == 'eglGetPlatformDisplay':
+                            query = 'epoxy_egl_version(EGL_NO_DISPLAY)'
+                            human_name += ' client'
+                        else:
+                            query = 'epoxy_conservative_egl_version()'
+
+                        condition = f'{query} >= {version}'
+
                 self.process_command_statements(command, condition, loader, human_name, human_name)
 
         for extension in reg.findall('extensions/extension'):
